POFile.js

```diff
@@ -617,7 +617,7 @@ function objectMap(object, visitor) {
  * @returns {String} the localized text of this file
  */
 POFile.prototype.localizeText = function(translations, locale) {
-    var l = new Locale(locale);
+    var l = new Locale((this.mapping && this.mapping.localeMap && this.mapping.localeMap[locale]) || this.project.getOutputLocale(locale));
```

Comment:
Please move functionality of getting mapped locale to a separate function. The condition is too big to be easy to understand.

```
var outputLocale = this.getOutputLocale(inputLocale); // this method has check for mapping.
var l = new Locale(outputLocale);
```